util.ts

```diff
@@ -1,18 +1,139 @@
+import { DataWriteOptions, TFile, TFolder, normalizePath } from "obsidian";
+import fs from 'fs/promises';
+import { PickedFile } from "filesystem";
+
+
 function escapeRegex(str: string): string {
 	return str.replace(/[.?*+^$[\]\\(){}|-]/g, '\\$&');
 }
 
-const ILLEGAL_CHARACTERS = '\\/:*?<>\"|';
-const ILLEGAL_FILENAME_RE = new RegExp('[' + escapeRegex(ILLEGAL_CHARACTERS) + ']', 'g');
+const ILLEGAL_FILENAME_CHARACTERS = '\\/:*?<>\"|';
+const ILLEGAL_FILENAME_RE = new RegExp('[' + escapeRegex(ILLEGAL_FILENAME_CHARACTERS) + ']', 'g');
+
+const ILLEGAL_HASHTAG_CHARACTERS = '\\:*?<>\"|!@#$%^&()+=\`\'~;,.';
```

Comment:
I would prefer if we consistently name these `tag` instead of mixing in `hashtag`